scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -460,6 +467,53 @@ protected NodeSet getRMNodes(Map<JobId, JobDescriptor> jobMap, int neededResourc
     }
 
     /**
+     * Create bindings which will be used by selection scripts for the given tasks
+     */
+    private Map<String, Serializable> createBindingsForSelectionScripts(InternalJob job, EligibleTaskDescriptor etd, InternalTask task) throws IOException, ClassNotFoundException {
+        Map<String, Serializable> bindings = new HashMap<>();
+        Map<String, Serializable> variables = new HashMap<>();
+
+
+        int resultSize = etd.getParents().size();
+        if (job.getType() == JobType.TASKSFLOW) {
+            // retrieve from the database the previous task results if available
+            if ((resultSize > 0) && task.handleResultsArguments()) {
+                List<TaskId> parentIds = new ArrayList<>(resultSize);
+                for (int i = 0; i < resultSize; i++) {
+                    parentIds.add(etd.getParents().get(i).getTaskId());
+                }
+                Map<TaskId, TaskResult> taskResults = schedulingService.getInfrastructure().getDBManager()
+                        .loadTasksResults(
+                                job.getId(), parentIds);
+                for (TaskResult taskResult : taskResults.values()) {
+                    if (taskResult.getPropagatedVariables() != null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)